azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/Subnet.java

```diff
@@ -55,13 +55,35 @@
             WithAttach<ParentT> withAddressPrefix(String cidr);
         }
 
+        /**
+         * The stage of the subnet definition allowing to specify the network security group to assign to the subnet.
+         * @param <ParentT> the parent type
+         */
+        interface WithNetworkSecurityGroup<ParentT> {
+            /**
+             * Assigns an existing network security group to this subnet.
+             * @param resourceId the resource ID of the network security group
+             * @return the next stage of the definition
+             */
+            WithAttach<ParentT> withExistingNetworkSecurityGroup(String resourceId);
+
+            /**
+             * Assigns an existing network security group to this subnet.
+             * @param nsg the network security group to assign
+             * @return the next stage of the definition
+             */
+            WithAttach<ParentT> withExistingNetworkSecurityGroup(NetworkSecurityGroup nsg);
```

Comment:
you are right.. yes, there is no other place where we have a creation shortcut for a top level resource from a child resource. I am fine with not having top level resource create from child resource, its reduces complexity.